app/main.py

```diff
@@ -1 +1,30 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name: str = name
+        self.health: int = health
+        self.hidden: bool = False
+        if self.health > 0:
+            Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, Hidden: {self.hidden}}}")
+
+    @classmethod
+    def update_alive(cls) -> None:
+        cls.alive = [animal for animal in cls.alive if animal.health > 0]
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, prey: "Animal") -> None:
+        if isinstance(prey, Herbivore) and not prey.hidden:
+            prey.health -= 50
+            if prey.health <= 0:
+                Animal.update_alive()
```

Comment:
The `update_alive` method should be called on the `Animal` class, not on an instance. The current implementation is correct in this regard, but ensure that the method is called whenever an animal's health changes to zero or below.